### ISSUE TITLE
CSharpWrappingCodeRefactorProvider encountered an error and has been disabled

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#61362]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/CSharpWrappingCodeRefactorProvider-encou/10034910)._

---
[severity:It bothers me. A fix would be nice] [regression] [worked-in:17.2 Preview 5.0]
![](https://i.imgur.com/fUE6uHy.png)

```
System.ArgumentOutOfRangeException : 'end' must not be less than 'start'
Parameter name: end
   at Microsoft.CodeAnalysis.Text.TextSpan.FromBounds(Int32 start,Int32 end)
   at Microsoft.CodeAnalysis.CSharp.Wrapping.SeparatedSyntaxList.CSharpParameterWrapper.PositionIsApplicable(SyntaxNode root,Int32 position,SyntaxNode declaration,Boolean containsSyntaxError,BaseParameterListSyntax listSyntax)
   at async Microsoft.CodeAnalysis.Wrapping.SeparatedSyntaxList.AbstractSeparatedSyntaxListWrapper`2.TryCreateComputerAsync[TListSyntax,TListItemSyntax]()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Wrapping.AbstractWrappingCodeRefactoringProvider.ComputeRefactoringsAsync()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync()
```

---
#### Feedback Bot on 5/9/2022, 05:53 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by Hamberfim** [comment]
SAME

System.ArgumentOutOfRangeException : 'end' must not be less than 'start'
Parameter name: end
   at Microsoft.CodeAnalysis.Text.TextSpan.FromBounds(Int32 start,Int32 end)
   at Microsoft.CodeAnalysis.CSharp.Wrapping.SeparatedSyntaxList.CSharpParameterWrapper.PositionIsApplicable(SyntaxNode root,Int32 position,SyntaxNode declaration,Boolean containsSyntaxError,BaseParameterListSyntax listSyntax)
   at async Microsoft.CodeAnalysis.Wrapping.SeparatedSyntaxList.AbstractSeparatedSyntaxListWrapper`2.TryCreateComputerAsync[TListSyntax,TListItemSyntax](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Wrapping.AbstractWrappingCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)

**Comment by jasonmalinowski** [comment]
I recently hit this too while I think I was editing a bunch of a generic arguments. I don't have a precise repro though.

@vatsalyaagrawal can we get somebody to investigate? We've already had a few hits of feedback and multiple bugs filed here too.

**Comment by jasonmalinowski** [comment]
So far looks maybe a few thousand hits too: https://prism.vsdata.io/failure/?query=c%3ACSharpParameterWrapper&eventType=fault&failureType=hits&failureHash=d167a20b-f315-7a2c-d1d2-9383a0e1ec25

**Comment by Bosch-Eli-Black** [comment]
@jasonmalinowski If we hit stuff like this but don't have a concrete repro, is there anything specific that we can provide to help out, like logs, or is that all automatically gathered by VS, anyway?

(Also, is it better to post these types of errors here or via the VS Feedback tool? I've been posting them via the VS Feedback tool, because I thought that posting them that way might include some extra helpful logging for the MS team, but when I post that way, the bugs usually get closed as "unreproducible", since I don't have a concrete repo)

Thanks 🙂

**Comment by jasonmalinowski** [comment]
@Bosch-Eli-Black: so submitting stuff via the VS Feedback tool does collect additional logs; whether those logs are actually _useful_ really depends on what the issue is. Performance issues with the feedback tool can attach ETW traces we can look at which can absolutely help. For outright crashes (i.e. not something like the yellow bar but VS outright crashing hard) it'll also help with dumps or give us some pointers to crash reports that your machine recently submitted. We also made some recent improvements where we may be able to better capture memory dumps for things that trigger these yellow bars, so this might be somewhere we want to try using that as well. That said, for issues like this it often comes down to a repro being needed, or us staring at the code and trying to think where it might have gone wrong.

Otherwise, "hey, I think this happened when I was typing something like X" never hurts. And even just a "I'm hitting this a lot, anybody know why?" at least sometimes gets the right attention too. Sometimes bugs just get lost in triage or telemetry systems misprocess data and we don't always notice until somebody goes and asks about a bug the old fashioned way.

**Comment by CyrusNajmabadi** [comment]
@ryzngard this would be another example to give the team about auto collecting dumps when enough hits have happened.

**Comment by Bosch-Eli-Black** [comment]
@jasonmalinowski Thanks, that's very helpful! 🙂

**Comment by Bosch-Eli-Black** [comment]
@jasonmalinowski Thanks, that's very helpful! 🙂

**Comment by EvilVir** [comment]
Same here :(

**Comment by DelWoodcock** [comment]
@CyrusNajmabadi 

Sorry no project, I was just doing some training videos and it kept popping up. I have restarted VS, then it popped up again.

The code was something like this:

            using (IDbConnection connection = new SqlConnection(connectionString))
            {
                var rows = await connection.QueryAsync<T>(storedProcedure, parameters,
                    commandType: CommandType.StoredProcedure);
                return rows.ToList();
            }
And I refactored it to this using refactoring, that is when I had the issue:

             using IDbConnection connection = new SqlConnection(connectionString);
            var rows = await connection.QueryAsync<T>(storedProcedure, parameters,
                commandType: CommandType.StoredProcedure);
            return rows.ToList();

**Comment by DelWoodcock** [comment]
@CyrusNajmabadi

I haven't had that issue lately, but it did do once, so I mad sure to kill all the VS processes , restarted VS and it did it again. Sorry I can't be of much help otherwise.

**Comment by CyrusNajmabadi** [comment]
when it did it again, can you include the code of hte method around your caret, and where your caret was?  that woudl be enough to solve htis.  thanks!

**Comment by DelWoodcock** [comment]
Well I restarted VS tried typing the following and put the cursor over the using statement to refactor and it didn't give me the error.  Sorry, but I will keep this thread in mind if it happen again.

        using (IDbConnection connection = new SqlConnection(connectionString))
        {
            var rows = await connection.QueryAsync<T>(storedProcedure, parameters,
                commandType: CommandType.StoredProcedure);
            return rows.ToList();
        }

**Comment by CyrusNajmabadi** [comment]
> give me the error. Sorry, but I will keep this thread in mind if it happen again.

Thanks!

Next time it happens, just capture the content of your current method, and what location the caret was at, and that shoudl be sufficient.  Thanks!

**Comment by swerve-city** [comment]
I am on version 17.2.5 and received the same error when following some general training.

`System.ArgumentOutOfRangeException : 'end' must not be less than 'start'
Parameter name: end
   at Microsoft.CodeAnalysis.Text.TextSpan.FromBounds(Int32 start,Int32 end)
   at Microsoft.CodeAnalysis.CSharp.Wrapping.SeparatedSyntaxList.CSharpParameterWrapper.PositionIsApplicable(SyntaxNode root,Int32 position,SyntaxNode declaration,Boolean containsSyntaxError,BaseParameterListSyntax listSyntax)
   at async Microsoft.CodeAnalysis.Wrapping.SeparatedSyntaxList.AbstractSeparatedSyntaxListWrapper`2.TryCreateComputerAsync[TListSyntax,TListItemSyntax](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Wrapping.AbstractWrappingCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)`

Closed and tried again and the error appears when I highlight a method I was in the process of typing out. Doesn't happen once I have the brackets in.

`public void UpsertRecord<T>`

**Comment by CyrusNajmabadi** [comment]
@swerve-city Awesome.  I repro the issue with that.  Working on a fix now.

**Comment by Bosch-Eli-Black** [comment]
Yay! ^_^

**Comment by twest820** [comment]
+1 from 17.3.0 during conversion a for to do while. Fix is assigned to 17.4.
```
System.ArgumentOutOfRangeException : 'end' must not be less than 'start'
Parameter name: end
   at Microsoft.CodeAnalysis.Text.TextSpan.FromBounds(Int32 start,Int32 end)
   at Microsoft.CodeAnalysis.CSharp.Wrapping.SeparatedSyntaxList.CSharpParameterWrapper.PositionIsApplicable(SyntaxNode root,Int32 position,SyntaxNode declaration,Boolean containsSyntaxError,BaseParameterListSyntax listSyntax)
   at async Microsoft.CodeAnalysis.Wrapping.SeparatedSyntaxList.AbstractSeparatedSyntaxListWrapper`2.TryCreateComputerAsync[TListSyntax,TListItemSyntax](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Wrapping.AbstractWrappingCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
```

**Comment by FlorianGrimm** [comment]
same problem VS 2022 17.3.3
public abstract class Abc<V, E> : Abc {
.. snip..
public static implicit operator Abc**<**
}

Error occours after typing <

**Comment by embix** [comment]
> The fix is in 17.4p1.

Saw this twice today in 17.3.5 - are we going to see a backport for 17.3? Or do we have to wait (a month?) for the 17.4 RTM?

**Comment by tyson-benson** [comment]
For the sake of anyone coming across this issue in the future. I experienced the issue when I added some rules to my `.editorconfig` file.

Specifically, I had added the following near the end of the file:
```
[*.{cs,vb}]
# Organize usings
dotnet_sort_system_directives_first = true
csharp_style_namespace_declarations = true:suggestion
dotnet_style_namespace_match_folder = true:suggestion
```

I cant remember where I got the above config from, but I believe these rules were invalid in some way. To resolve the issue, I stripped my `.editorconfig` file contents down to bare minimum. And then instead of modifying it by hand (in a text editor), I used the Visual Studio UI for editing the .editorconfig file. After this, the refactoring providers no longer failed on opening the affected solution.
